wallet.js

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import hathorLib from '@hathor/wallet-lib';
+
+import EventEmitter from 'events';
+
+/**
+ * This is a Wallet that is supposed to be simple to be used by a third-party app.
+ *
+ * This class handles all the details of syncing, including receiving the same transaction
+ * multiple times from the server. It also keeps the balance of the tokens updated.
+ *
+ * It has the following states:
+ * - CLOSED: When it is disconnected from the server.
+ * - CONNECTING: When it is connecting to the server.
+ * - SYNCING: When it has connected and is syncing the transaction history.
+ * - READY: When it is ready to be used.
+ *
+ * You can subscribe for the following events:
+ * - state: When the state of the Wallet changes.
+ * - new-tx: When a new tx arrives.
+ * - update-tx: When a known tx is updated. Usually, it happens when one of its outputs is spent.
+ **/
+class Wallet extends EventEmitter {
+  constructor({ network, server, seed }) {
+    super();
+
+    hathorLib.network.setNetwork(network);
+    this.network = network;
+
+    this.state = Wallet.CLOSED;
+    this.serverInfo = null;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
+    this.onAddressesLoaded = this.onAddressesLoaded.bind(this);
```

Comment:
This is not doing anything. Is it just for future use?

Reply:
Now it is propagating the event. Thanks!